app/controllers/application_controller.rb

```diff
@@ -1,2 +1,10 @@
 class ApplicationController < ActionController::API
+  before_action :configure_permitted_parameters, if: :devise_controller?
+  before_action :authenticate_api_v1_user!
+
+  protected
+
+  def configure_permitted_parameters
+    devise_parameter_sanitizer.permit(:sign_up, keys: %i[name email password password_confirmation])
+  end
 end
```

Comment:
Please add a `current_ability` method in the application controller. Since User is namespaced under Api and V1 in the routes, the name of the `current_user` method has changed to `current_api_v1_user` which is causing errors in the CanCanCan Ability class. Add a `current_ability` method in the application controller under `protected` like this to fix this error
```suggestion
class ApplicationController < ActionController::API
  before_action :configure_permitted_parameters, if: :devise_controller?
  before_action :authenticate_api_v1_user!

  protected
 
  def current_ability
    @current_ability ||= Ability.new(current_api_v1_user)
  end
  def configure_permitted_parameters
    devise_parameter_sanitizer.permit(:sign_up, keys: %i[name email password password_confirmation])
  end
end
```